app/main.py

```diff
@@ -2,24 +2,84 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.public_name = name
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: type) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if self.min_amount <= value <= self.max_amount:
+            setattr(instance, self.protected_name, value)
+            if not hasattr(instance, "access"):
+                instance.access = True
+        else:
+            print(f"{instance.slide_name}: access is not granted"
+                  f" because of {self.public_name}({value}) "
+                  f"is not in range({self.min_amount}-{self.max_amount})")
+            if not hasattr(instance, "access") or instance.access:
+                instance.access = False
 
 
 class Visitor:
-   pass
+    def __init__(self,
+                 name: str,
+                 age: int,
+                 weight: int,
+                 height: int
+                 ) -> None:
+        self.name = name
+        self.age = age
+        self.height = height
+        self.weight = weight
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(self,
+                 age: int,
+                 weight: int,
+                 height: int,
+                 slide_name: str,
+                 ) -> None:
+        self.slide_name = slide_name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    height = IntegerRange(80, 120)
+    weight = IntegerRange(20, 50)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(14, 60)
+    height = IntegerRange(120, 220)
+    weight = IntegerRange(50, 120)
 
 
 class Slide:
-    pass
+    def __init__(self,
+                 name: str,
+                 limitation_class: type
```

Comment:
Indicate that you expect to obtain a subclass of the `SlideLimitationValidator` for `limitation_class`:
```suggestion
                 limitation_class: type[SlideLimitationValidator]
```